lib/metrics.js

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+var client = require("prom-client");
+
+class Metrics {
+  constructor() {
+    this.register = new client.Registry();
+    client.collectDefaultMetrics({ register: this.register });
+
+    this.apiRouteGetCount = new client.Counter({
+      name: "api_route_get",
+      help: "Count of API route get requests",
+      registers: [this.register],
+    });
+
+    this.apiRouteAddCount = new client.Counter({
+      name: "api_route_add",
+      help: "Count of API route add requests",
+      registers: [this.register],
+    });
+
+    this.apiRouteDeleteCount = new client.Counter({
+      name: "api_route_delete",
+      help: "Count of API route delete requests",
+      registers: [this.register],
+    });
+
+    this.findTargetForReqSummary = new client.Summary({
+      name: "find_target_for_req",
+      help: "Summary of find target requests",
+      registers: [this.register],
+    });
+
+    this.lastActivityUpdatingSummary = new client.Summary({
+      name: "last_activity_updating",
+      help: "Summary of last activity updating requests",
+      registers: [this.register],
+    });
+
+    this.requestsWsCount = new client.Counter({
+      name: "requests_ws",
+      help: "Count of websocket requests",
+      registers: [this.register],
+    });
+
+    this.requestsWebCount = new client.Counter({
+      name: "requests_web",
+      help: "Count of web requests",
+      registers: [this.register],
+    });
+
+    this.requestsProxyCount = new client.Counter({
+      name: "requests_proxy",
+      help: "Count of proxy requests",
+      labelNames: ["status"],
+      registers: [this.register],
+    });
+
+    this.requestsApiCount = new client.Counter({
+      name: "requests_api",
+      help: "Count of API requests",
+      labelNames: ["status"],
+      registers: [this.register],
+    });
+  }
+
+  render(res) {
+    return this.register.metrics().then((s) => {
+      res.writeHead(200, { "Content-Type": this.register.contentType });
```

Comment:
Not feeling confident about potential issues or not, but should there be a that = this statement and a reference to that within here?

Reply:
The `that = this` pattern is generally not needed anymore *if* you use the newer `() => {}` arrow-function syntax as is done here, instead of `function() { ... }`. For example:

```javascript
// old way

var that = this;
asyncFunction().then(function (result) {
  // callign a regular `function` results in a different `this` context variable, so `this.method` won't work
  that.method();
});

// new way

asyncFunction().then(() => {
  this.method(); // arrow functions *inherit* `this` context, so we can skip that = this rigamarole
})
```

CHP predates the general availability of this feature, though, so it's still use it in a number of places, but it can be phased out, and doesn't need to be added to new code.

Reply:
Thanks for helping me learn about this Min!